uitests/tests/org.jboss.tools.fuse.ui.bot.tests/src/org/jboss/tools/fuse/ui/bot/tests/CamelRestDSLFromWSDLWizardTest.java

```diff
@@ -0,0 +1,317 @@
+import org.jboss.tools.fuse.reddeer.utils.ProjectFactory;
+import org.jboss.tools.fuse.reddeer.wizard.CamelRestDSLFromWSDLAdvancedPage;
+import org.jboss.tools.fuse.reddeer.wizard.CamelRestDSLFromWSDLFirstPage;
+import org.jboss.tools.fuse.reddeer.wizard.CamelRestDSLFromWSDLWizard;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * Tests <i>Camel Rest DSL from WSDL</i> wizard<br/>
+ * <b>Quick tests:</b><br/>
+ * First wizard page:
+ * <ul>
+ * <li>testExampleWSDLFile</li>
+ * <li>testEmptyWSDLFile</li>
+ * <li>testInvalidWSDLFile</li>
+ * <li>testEnablingDisablingButtons</li>
+ * </ul>
+ * Second wizard page:
+ * <ul>
+ * <li>testDestinationCamelFolder</li>
+ * <li>testDestinationJavaFolder</li>
+ * </ul>
+ * 
+ * @author djelinek
+ *
+ */
+@CleanWorkspace
+@OpenPerspective(FuseIntegrationPerspective.class)
+@RunWith(RedDeerSuite.class)
+public class CamelRestDSLFromWSDLWizardTest {
+
+	public static final String WSDL_FILE_EXAMPLE_PATH = "resources/wsdl2rest/WSDLFileExample.wsdl";
+	public static final String WSDL_FILE_EXAMPLE_ERROR_PATH = "resources/wsdl2rest/WSDLFileExample";
+	public static final String WSDL_FILE_EMPTY_PATH = "resources/wsdl2rest/WSDLFileEmpty.wsdl";
+	public static final String WSDL_FILE_INVALID_PATH = "resources/wsdl2rest/WSDLFileInvalid.wsdl";
+	public static final String PROJECT_NAME = "wsdl2rest";
+
+	private String path;
+	private CamelRestDSLFromWSDLWizard wizard;
+	private CamelRestDSLFromWSDLFirstPage firstPage;
+	private CamelRestDSLFromWSDLAdvancedPage secondPage;
+
+	@Before
+	public void initSetup() {
+		new CleanErrorLogRequirement().fulfill();
+		ProjectFactory.newProject(PROJECT_NAME).deploymentType(STANDALONE).runtimeType(KARAF).template(CBR_SPRING)
+				.create();
+
+		// should have ensure pre-filling 'Destination Project' wizard field
+		new ProjectExplorer().selectProjects(PROJECT_NAME);
+		wizard = new CamelRestDSLFromWSDLWizard();
+		wizard.open();
+		firstPage = new CamelRestDSLFromWSDLFirstPage(wizard);
+	}
+
+	@After
+	public void cleanWorkspace() {
+		new CleanWorkspaceRequirement().fulfill();
+	}
+
+	/**
+	 * <p>
+	 * Test tries to 'Finish' wizard with path to 'example' of WSDL file
+	 * </p>
+	 * <b>Steps:</b>
+	 * <ol>
+	 * <li>Verifies that 'Destination Project' field was pre-filled</li>
+	 * <li>Set 'wrong' path to 'example' WSDL file</li>
+	 * <li>Verifies that wizard 'Finish' button is disabled</li>
+	 * <li>Set 'valid' path to 'example' WSDL file</li>
+	 * <li>Verifies that wizard 'Finish' button is enabled</li>
+	 * <li>Tries to 'Finish' wizard</li>
+	 * <li>Verifies 'Error log' to 'fuse' error</li>
+	 * </ol>
+	 */
+	@Test
+	public void testExampleWSDLFile() {
+		assertDestinationProjectFieldCompetion();
+
+		path = ResourceHelper.getResourceAbsolutePath(Activator.PLUGIN_ID, WSDL_FILE_EXAMPLE_ERROR_PATH);
+		firstPage.setTextWSDLFile("file:" + path);
+		AbstractWait.sleep(TimePeriod.MEDIUM);
```

Comment:
Couldn't you use some smarter wait condition here? JobIsRunning? It is just a question.

Reply:
The validation of fields is not probably related to JobIsRunning condition, because tests fails if I used that.. I can create some new wait condition or use for example "new WaitUntil(new ControlIsEnabled(new PushButton("Finish")));"...

Reply:
`new WaitUntil(new ControlIsEnabled(new PushButton("Finish")));`
looks good to me ;-)